utensor_cgen/ir/instr.py

```diff
@@ -0,0 +1,53 @@
+class MemoryAllocation:
+    _store_name = "address"
+    
+    def __init__(self):
+        self._address_map = {}
+
+    def __setattr__(self, attr, value):
+        if attr == '_address_map':
+            return super().__setattr__(attr, value)
+        self._address_map[attr] = value
+    
+    def __getattr__(self, att):
+        return self._address_map[att]
+
+class DataManager:
+    InfoCenterMap = {
+        MemoryAllocation._store_name: MemoryAllocation
+    }
+    def __init__(self, datas):
+        self.StorageCenter = {}
+        for data_name in datas:
+            data_cls = self.InfoCenterMap.get(data_name, None)
+            if data_cls is None:
+                raise ValueError("Unknown transformation method: {}".format(data_name))
+            datastorage = data_cls()
+            self.StorageCenter.update({data_name: datastorage})
+
+    @classmethod
+    def register_datamap(cls, data_cls, overwrite=False):
```

Comment:
why do you need this `overwrite` arguments?
Looks like it's of no use in the function body.